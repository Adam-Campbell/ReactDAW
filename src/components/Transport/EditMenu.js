import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import * as ActionCreators from '../../actions';

export const EditMenu = props => {
    return (
        <div 
            className="menu__container" 
            ref={props.containerRef}
            onClick={props.toggleMenu}
        >
            <p 
                className="menu__label"
            >Edit</p>
            <ul 
                className={`menu__list ${props.isOpen ? 'menu__list--visible' : ''}`}
            >
                <li 
                    className="menu__item"
                >
                    <p className="menu__item-text">Undo</p>
                </li>
                <li className="menu__item">
                    <p className="menu__item-text">Redo</p>
                </li>
            </ul>
        </div>
    );
}

EditMenu.propTypes = {
    isOpen: PropTypes.bool.isRequired,
    toggleMenu: PropTypes.func.isRequired,
    containerRef: PropTypes.object.isRequired
}

export default connect(undefined)(EditMenu);