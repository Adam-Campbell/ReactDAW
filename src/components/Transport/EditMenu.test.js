import React from 'react';
import { EditMenu } from './EditMenu';
import { shallow } from 'enzyme';

test('renders closed state correctly', () => {
    const component = shallow(
        <EditMenu 
            isOpen={false}
            toggleMenu={jest.fn()}
            containerRef={React.createRef()}
        />
    );
    expect(component).toMatchSnapshot();
});

test('renders open state correctly', () => {
    const component = shallow(
        <EditMenu 
            isOpen={true}
            toggleMenu={jest.fn()}
            containerRef={React.createRef()}
        />
    );
    expect(component).toMatchSnapshot(); 
});