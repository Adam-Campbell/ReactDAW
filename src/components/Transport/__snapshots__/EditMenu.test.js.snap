// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders closed state correctly 1`] = `
<div
  className="menu__container"
  onClick={[MockFunction]}
>
  <p
    className="menu__label"
  >
    Edit
  </p>
  <ul
    className="menu__list "
  >
    <li
      className="menu__item"
    >
      <p
        className="menu__item-text"
      >
        Undo
      </p>
    </li>
    <li
      className="menu__item"
    >
      <p
        className="menu__item-text"
      >
        Redo
      </p>
    </li>
  </ul>
</div>
`;

exports[`renders open state correctly 1`] = `
<div
  className="menu__container"
  onClick={[MockFunction]}
>
  <p
    className="menu__label"
  >
    Edit
  </p>
  <ul
    className="menu__list menu__list--visible"
  >
    <li
      className="menu__item"
    >
      <p
        className="menu__item-text"
      >
        Undo
      </p>
    </li>
    <li
      className="menu__item"
    >
      <p
        className="menu__item-text"
      >
        Redo
      </p>
    </li>
  </ul>
</div>
`;
